stdlib/LinearAlgebra/src/symmetriceigen.jl

```diff
@@ -309,19 +309,19 @@ function eigen(A::Hermitian{Complex{T}, <:Tridiagonal}; kwargs...) where {T}
             E = dl
             Er = abs.(E)
         end
-        S = Vector{Complex{eigtype(T)}}(undef, N)
+        S = Vector{eigtype(eltype(A))}(undef, N)
         S[1] = 1
         for i ∈ 1:N-1
             S[i+1] = iszero(Er[i]) ? oneunit(eltype(S)) : S[i] * sign(E[i])
         end
-        B = SymTridiagonal(real(d), Er)
+        B = SymTridiagonal(float.(real.(d)), Er)
```

Comment:
If used without broadcasting, then this is a no-op if the array has already float eltype. Or do you need a freshly allocated vector here? Same below.

Reply:
This is to convert the complex vector `d` to a real one with a floating-point `eltype`, so a fresh allocation will be necessary

Reply:
I know, it's just a nitpick, but `float(real(d))` will allocate if necessary, and will avoid allocations otherwise.

Reply:
Won't this allocate twice in this case, e.g. for a `Vector{Complex{Int}}`?

Reply:
> Won't this allocate twice in this case, e.g. for a `Vector{Complex{Int}}`?

Yeah, that's true. Sorry for the noise.